src/main/java/evm/ContractTransaction.java

```diff
@@ -0,0 +1,138 @@
+package evm;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.vm.DataWord;
+import org.ethereum.vm.client.*;
+import org.ethereum.vm.util.ByteArrayUtil;
+import org.ethereum.vm.util.HashUtil;
+import org.ethereum.vm.util.HexUtil;
+import static org.ethereum.vm.util.ByteArrayUtil.merge;
+import org.apache.log4j.Logger;
+
+public class ContractTransaction extends vmbase {
+
+    public final BigInteger premine = BigInteger.valueOf(1L).multiply(Unit.ETH); // each account has 1 ether
+    public Transaction transaction;
+    public Block block;
+    final static Logger logger = Logger.getLogger(ContractTransaction.class);
+
+    public void setup() {
+        super.setup();
+        transaction = new TransactionMock(false, caller, address, 0, value, data, gas, gasPrice);
+        block = new BlockMock(number, prevHash, coinbase, timestamp, gasLimit);
+        repository.addBalance(origin, premine);
+        repository.addBalance(caller, premine);
+        repository.addBalance(address, premine);
+        repository.addBalance(coinbase, premine);
+    }
+
+     /**
+	 * This function reads a contract by taking the name of contract
+     *
+	 * @param contractLocation is the location of the stored contract
+     * @param address represents the account address
+     * @param nonce is a random, one-time, whole number
+     * @param gas refers to the cost necessary to perform a transaction
+	 */
+    public byte[] createContract(String contractLocation, byte[] address, long nonce, long gas) throws IOException {
+        return createContract(readContract(contractLocation), new byte[0], address, nonce, gas);
+    }
+
+    
+    /**
+	 * This function reads a contract by taking the name of contract
+     *
+	 * @param fileName is the contract name 
+	 */
+    public byte[] readContract(String fileName) throws IOException {
+        String path3 = fileName;
```

Comment:
we can use the `fileName`, no need for additional local variable.

Reply:
fixed.